integrationTests/vm/arwen/arwenvm/deployment/upgrade_test.go

```diff
@@ -0,0 +1,105 @@
+package deployment
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm/arwen"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScUpgradeShouldManageCorrectlyTheCodeMetadata(t *testing.T) {
+	senderAddressBytes := []byte("12345678901234567890123456789012")
+	senderNonce := uint64(0)
+	senderBalance := big.NewInt(100000000)
+
+	testContext, err := vm.CreatePreparedTxProcessorAndAccountsWithVMs(
+		senderNonce,
+		senderAddressBytes,
+		senderBalance,
+		config.EnableEpochs{
+			IsPayableBySCEnableEpoch: 1,
+		},
+	)
+	require.Nil(t, err)
+	defer testContext.Close()
+
+	t.Run("payable by SC is not active, code metadata is not parsed", func(t *testing.T) {
```

Comment:
can you add some comment that it is needed for backwards compatibility?

Reply:
done